tests/datasets/processors/test_generic_metrics_processor.py

```diff
@@ -20,11 +21,37 @@
         ),
     ],
 )
-def test__should_process(
+def test__counters_should_process(
     message: Mapping[str, Any], expected_output: Mapping[str, Any]
 ) -> None:
-    processor = GenericCountersMetricsProcessor()
-    assert processor._should_process(message) == expected_output
+    counters_processor = GenericCountersMetricsProcessor()
+    assert counters_processor._should_process(message) == expected_output
+
+
+@pytest.mark.parametrize(
+    "message, expected_output",
+    [
+        pytest.param(
+            {
+                "type": "g",
+                "metric_id": 1,
+                "value": {"min": 0, "max": 3, "sum": 3, "count": 2, "last": 3},
+            },
+            True,
+            id="gauge",
+        ),
+        pytest.param(
+            {"type": "d", "metric_id": 2, "value": 20},
+            False,
+            id="distribution",
```

Comment:
Is there a reason you added distribution to the gauge metrics processor test? I would imagine we would never want to do this in production.

Reply:
Oh, I think I mainly added it just to check that the `gauges` processor does not process a message with any other type other than `g`. 

I can remove it if that seems redundant.